osu.Game/Rulesets/Edit/Checks/CheckMutedObjects.cs

```diff
@@ -0,0 +1,147 @@
+        {
+            foreach (var hitObject in context.Beatmap.HitObjects)
+            {
+                // Worth keeping in mind: The samples of an object always play at its end time.
+                // Objects like spinners have no sound at its start because of this, while hold notes have nested objects to accomplish this.
+                foreach (var nestedHitObject in hitObject.NestedHitObjects)
+                {
+                    foreach (var issue in getVolumeIssues(hitObject, nestedHitObject))
+                        yield return issue;
+                }
+
+                foreach (var issue in getVolumeIssues(hitObject))
+                    yield return issue;
+            }
+        }
+
+        private IEnumerable<Issue> getVolumeIssues(HitObject hitObject, HitObject sampledHitObject = null)
+        {
+            sampledHitObject ??= hitObject;
+            if (!sampledHitObject.Samples.Any())
+                yield break;
+
+            // Samples that allow themselves to be overridden by control points have a volume of 0.
+            int maxVolume = sampledHitObject.Samples.Max(sample => sample.Volume > 0 ? sample.Volume : sampledHitObject.SampleControlPoint.SampleVolume);
+            double samplePlayTime = sampledHitObject.GetEndTime();
+
+            EdgeType edgeType = getEdgeAtTime(hitObject, samplePlayTime);
+            // We only care about samples played on the edges of objects, not ones like spinnerspin or slidertick.
+            if (edgeType == EdgeType.None)
+                yield break;
+
+            string postfix = hitObject is IHasDuration ? edgeType.ToString().ToLower() : null;
+
+            if (maxVolume <= muted_threshold)
+            {
+                if (edgeType == EdgeType.Head)
+                    yield return new IssueTemplateMutedActive(this).Create(hitObject, maxVolume / 100f, sampledHitObject.GetEndTime(), postfix);
+                else
+                    yield return new IssueTemplateMutedPassive(this).Create(hitObject, maxVolume / 100f, sampledHitObject.GetEndTime(), postfix);
+            }
+            else if (maxVolume <= low_volume_threshold && edgeType == EdgeType.Head)
+            {
+                yield return new IssueTemplateLowVolumeActive(this).Create(hitObject, maxVolume / 100f, sampledHitObject.GetEndTime(), postfix);
+            }
+        }
+
+        private EdgeType getEdgeAtTime(HitObject hitObject, double time)
+        {
+            if (Precision.AlmostEquals(time, hitObject.StartTime, 1f))
+                return EdgeType.Head;
+            if (Precision.AlmostEquals(time, hitObject.GetEndTime(), 1f))
+                return EdgeType.Tail;
+
+            if (hitObject is IHasRepeats hasRepeats)
+            {
+                double spanDuration = hasRepeats.Duration / hasRepeats.SpanCount();
+                if (Precision.AlmostEquals((time - hitObject.StartTime) % spanDuration, 0f, 1f))
```

Comment:
I generally do not recommend using floating-point modulo because it can have unexpected properties:

```
12.1 % 0.1 = 0.09999999999999898
 0.2 % 0.1 = 0
-0.1 % 0.1 = -0
-0.3 % 0.1 = -0.09999999999999998
```

In this case there's nothing that guarantees you won't get a value just ever so slightly below `spanDuration`. I'd recommend using `(time - hitObject.StartTime) / spanDuration` instead and probably doing iterative `AlmostEquals()` checks with 1, 2, 3, ... until that division result is exceeded.

Reply:
Ended up checking if the result of the division is a whole number:
```cs
double spans = (time - hitObject.StartTime) / spanDuration;

if (Precision.AlmostEquals(spans, Math.Ceiling(spans)) || Precision.AlmostEquals(spans, Math.Floor(spans)))
    return EdgeType.Repeat;
```
Should accomplish the same thing without iteration. The acceptable difference on `AlmostEquals` may be a bit too small as-is though, would depend on `spanDuration`.

Reply:
I think that for something like this a mix of relative and absolute error would be the absolute best (something like `Math.Max(1e-3, 1 / spanDuration)`) when it comes to the precision, but for simplicity going with something constant like `1e-3` might be good enough. Not sure I'd leave the currently-used default of `1e-7` as it seems a bit on the smaller side.

Reply:
Ended up going for `1 / spanDuration` (1 ms) for consistency with the acceptable difference of head/tail edges. Likelihood that `spanDuration` is so large that the resulting acceptable difference is less than E-7 is quite low in any realistic case, I think.

Reply:
Sure, I'd say that also makes sense.